app/controllers/opportunities_controller.rb

```diff
@@ -0,0 +1,71 @@
+class OpportunitiesController < ApplicationController
+  before_action :set_opportunity, only: [:edit, :update, :show, :destroy]
+  before_action :require_company_owner, only: [:edit, :update, :destroy]
+
+  def index
+    @opportunities = Opportunity.all
+  end
+
+  def new
+    # redirect to root path if user has no company yet
+    if current_user.companies.empty?
+      redirect_to new_company_path
+      flash[:danger] = "You must create a company before adding an internship"
```

Comment:
Should sequentially add the flash before the redirect just for even readability of the order.